### PR TITLE
Map a lot of realms stuff

### PR DESCRIPTION
**Pull request by FlowArg**
Hello, 

There are a lot of `com.mojang.realmsclient.*` mapped stuff in this PR.

Have a nice day

### CODE REVIEW

File: data/com/mojang/realmsclient/dto/UploadInfo.mapping [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	METHOD createRequest (Ljava/lang/String;)Ljava/lang/String;
+		ARG 0 token
+	METHOD ensureEndpointSchema (Ljava/lang/String;Ljava/util/regex/Matcher;)Ljava/lang/String;
+		ARG 0 uri
+		ARG 1 matcher
+	METHOD parse (Ljava/lang/String;)Lcom/mojang/realmsclient/dto/UploadInfo;
+		ARG 0 json
+	METHOD selectPortOrDefault (II)I
+		ARG 0 firstPort
+		ARG 1 secondPort
```
Comment (alcatrazEscapee) [inline data/com/mojang/realmsclient/dto/UploadInfo.mapping]:
```suggestion
		ARG 0 port
		ARG 1 defaultPort
```

Seeing as the method name indicates "port or default"

File: data/com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen.mapping [hunk 2]
```diff
@@ -1,4 +1,21 @@
 CLASS com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen
+	METHOD <init> (Lcom/mojang/realmsclient/exception/RealmsServiceException;Lnet/minecraft/client/gui/screens/Screen;)V
+		ARG 1 error
+		ARG 2 nextScreen
+	METHOD <init> (Lnet/minecraft/network/chat/Component;Lnet/minecraft/client/gui/screens/Screen;)V
+		ARG 1 error
+		ARG 2 nextScreen
+	METHOD <init> (Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/client/gui/screens/Screen;)V
+		ARG 1 errorLine1
+		ARG 2 errorLine2
```
Comment (alcatrazEscapee) [inline data/com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen.mapping]:
```suggestion
		ARG 1 line1
		ARG 2 line2
```
from field names

File: data/com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen.mapping [hunk 3]
```diff
@@ -1,4 +1,21 @@
...
+		ARG 3 nextScreen
+	METHOD errorMessage (Lcom/mojang/realmsclient/exception/RealmsServiceException;)V
+		ARG 1 exception
+	METHOD errorMessage (Lnet/minecraft/network/chat/Component;)V
+		ARG 1 error
+	METHOD errorMessage (Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;)V
+		ARG 1 errorLine1
+		ARG 2 errorLine2
```
Comment (alcatrazEscapee) [inline data/com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen.mapping]:
```suggestion
		ARG 1 line1
		ARG 2 line2
```
from field names

File: data/com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen.mapping [hunk 4]
```diff
@@ -1,4 +1,21 @@
...
+		ARG 1 error
+		ARG 2 nextScreen
+	METHOD <init> (Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/client/gui/screens/Screen;)V
+		ARG 1 errorLine1
+		ARG 2 errorLine2
+		ARG 3 nextScreen
+	METHOD errorMessage (Lcom/mojang/realmsclient/exception/RealmsServiceException;)V
+		ARG 1 exception
+	METHOD errorMessage (Lnet/minecraft/network/chat/Component;)V
+		ARG 1 error
```
Comment (alcatrazEscapee) [inline data/com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen.mapping]:
```suggestion
		ARG 1 line2
```
from field names, seeing as the first line is set automatically in this case.

File: data/com/mojang/realmsclient/gui/screens/RealmsPendingInvitesScreen.mapping [hunk 5]
```diff
@@ -8,6 +10,11 @@ CLASS com/mojang/realmsclient/gui/screens/RealmsPendingInvitesScreen
 		ARG 2 mouseX
 		ARG 3 mouseY
 		ARG 4 partialTicks
+	METHOD renderMousehoverTooltip (Lcom/mojang/blaze3d/vertex/PoseStack;Lnet/minecraft/network/chat/Component;II)V
+		ARG 1 poseStack
+		ARG 2 toolTip
```
Comment (alcatrazEscapee) [inline data/com/mojang/realmsclient/gui/screens/RealmsPendingInvitesScreen.mapping:15]:
```suggestion
		ARG 2 tooltip
```

File: data/net/minecraft/advancements/AdvancementProgress.mapping [hunk 6]
```diff
@@ -20,12 +20,3 @@ CLASS net/minecraft/advancements/AdvancementProgress
...
 		ARG 2 requirements
-	CLASS Serializer
-		METHOD deserialize (Lcom/google/gson/JsonElement;Ljava/lang/reflect/Type;Lcom/google/gson/JsonDeserializationContext;)Ljava/lang/Object;
-			ARG 1 json
-			ARG 2 typeOfT
-			ARG 3 context
-		METHOD serialize (Ljava/lang/Object;Ljava/lang/reflect/Type;Lcom/google/gson/JsonSerializationContext;)Lcom/google/gson/JsonElement;
-			ARG 1 src
-			ARG 2 typeOfSrc
-			ARG 3 context
```
Comment (alcatrazEscapee) [inline data/net/minecraft/advancements/AdvancementProgress.mapping]: This is already being handled by updated compass with #67 (as it's meant to be mapping the target, not the bouncer.) So omit it from this PR.

**Review by alcatrazEscapee — CHANGES_REQUESTED** [review]
A couple suggestions, otherwise looks good standards wise.

File: data/com/mojang/realmsclient/gui/screens/RealmsPendingInvitesScreen.mapping (hunk 5 above)
Comment (alcatrazEscapee) [reply data/com/mojang/realmsclient/gui/screens/RealmsPendingInvitesScreen.mapping:15]: Still relevant

File: data/net/minecraft/advancements/AdvancementProgress.mapping (hunk 6 above)
Comment (alcatrazEscapee) [reply data/net/minecraft/advancements/AdvancementProgress.mapping]: Still relevant.

**Review by alcatrazEscapee — DISMISSED** [review]

File: data/com/mojang/realmsclient/dto/RealmsServer.mapping [hunk 7]
```diff
@@ -0,0 +1,32 @@
+CLASS com/mojang/realmsclient/dto/RealmsServer
+	METHOD cloneSlots (Ljava/util/Map;)Ljava/util/Map;
+		ARG 1 slots
+	METHOD equals (Ljava/lang/Object;)Z
+		ARG 1 other
+	METHOD getState (Ljava/lang/String;)Lcom/mojang/realmsclient/dto/RealmsServer$State;
+		ARG 0 state
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/dto/RealmsServer.mapping]:
```suggestion
		ARG 0 name
```

File: data/com/mojang/realmsclient/dto/RealmsServer.mapping [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+	METHOD getWorldName (I)Ljava/lang/String;
+		ARG 1 slot
+	METHOD getWorldType (Ljava/lang/String;)Lcom/mojang/realmsclient/dto/RealmsServer$WorldType;
+		ARG 0 worldType
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/dto/RealmsServer.mapping]:
```suggestion
		ARG 0 name
```

File: data/com/mojang/realmsclient/dto/RealmsWorldOptions.mapping [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+		ARG 5 spawnProtection
+		ARG 6 commandBlocks
+		ARG 7 difficulty
+		ARG 8 gameMode
+		ARG 9 forceGameMode
+		ARG 10 slotName
+	METHOD getDefaultSlotName (I)Ljava/lang/String;
+		ARG 1 slotId
+	METHOD getSlotName (I)Ljava/lang/String;
+		ARG 1 slotId
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/dto/RealmsWorldOptions.mapping]: From passed field names. 

```suggestion
	METHOD getDefaultSlotName (I)Ljava/lang/String;
		ARG 1 slotIndex
	METHOD getSlotName (I)Ljava/lang/String;
		ARG 1 slotIndex
```

File: data/com/mojang/realmsclient/dto/Subscription.mapping [hunk 10]
```diff
@@ -0,0 +1,5 @@
+CLASS com/mojang/realmsclient/dto/Subscription
+	METHOD parse (Ljava/lang/String;)Lcom/mojang/realmsclient/dto/Subscription;
+		ARG 0 json
+	METHOD typeFrom (Ljava/lang/String;)Lcom/mojang/realmsclient/dto/Subscription$SubscriptionType;
+		ARG 0 type
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/dto/Subscription.mapping]:
```suggestion
		ARG 0 name
```

File: data/com/mojang/realmsclient/exception/RealmsServiceException.mapping [hunk 11]
```diff
@@ -0,0 +1,10 @@
+CLASS com/mojang/realmsclient/exception/RealmsServiceException
+	METHOD <init> (ILjava/lang/String;ILjava/lang/String;)V
+		ARG 1 httpResultCode
+		ARG 2 message
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/exception/RealmsServiceException.mapping]: More specific, uses field name. 

```suggestion
		ARG 2 httpResponseContent
```

File: data/com/mojang/realmsclient/exception/RealmsServiceException.mapping [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+		ARG 3 errorCode
+		ARG 4 errorMsg
+	METHOD <init> (ILjava/lang/String;Lcom/mojang/realmsclient/client/RealmsError;)V
+		ARG 1 httpResultCode
+		ARG 2 message
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/exception/RealmsServiceException.mapping]:
```suggestion
		ARG 2 httpResponseContent
```

File: data/com/mojang/realmsclient/gui/screens/RealmsBackupScreen.mapping [hunk 13]
```diff
@@ -20,12 +32,19 @@ CLASS com/mojang/realmsclient/gui/screens/RealmsBackupScreen
...
+		METHOD addEntry (Lcom/mojang/realmsclient/dto/Backup;)V
+			ARG 1 backup
 		METHOD mouseClicked (DDI)Z
 			ARG 1 mouseX
 			ARG 3 mouseY
 			ARG 5 button
 		METHOD renderBackground (Lcom/mojang/blaze3d/vertex/PoseStack;)V
 			ARG 1 matrixStack
+		METHOD selectInviteListItem (I)V
+			ARG 1 backupId
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/gui/screens/RealmsBackupScreen.mapping]: From nearby mappings. 

```suggestion
			ARG 1 index
```

File: data/com/mojang/realmsclient/gui/screens/RealmsBrokenWorldScreen.mapping [hunk 14]
```diff
@@ -1,4 +1,16 @@
 CLASS com/mojang/realmsclient/gui/screens/RealmsBrokenWorldScreen
+	METHOD <init> (Lnet/minecraft/client/gui/screens/Screen;Lcom/mojang/realmsclient/RealmsMainScreen;JZ)V
+		ARG 1 lastScreen
+		ARG 2 mainScreen
+		ARG 3 serverId
+		ARG 5 isMinigame
+	METHOD downloadWorld (I)V
+		ARG 1 slotId
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/gui/screens/RealmsBrokenWorldScreen.mapping]:
```suggestion
		ARG 1 slotIndex
```

File: data/com/mojang/realmsclient/gui/screens/RealmsBrokenWorldScreen.mapping [hunk 15]
```diff
@@ -1,4 +1,16 @@
...
+	METHOD drawSlotFrame (Lcom/mojang/blaze3d/vertex/PoseStack;IIIIZLjava/lang/String;IJLjava/lang/String;Z)V
+		ARG 1 poseStack
+		ARG 8 slotId
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/gui/screens/RealmsBrokenWorldScreen.mapping]:
```suggestion
		ARG 8 slotIndex
```

File: data/com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen.mapping [hunk 16]
```diff
@@ -1,4 +1,21 @@
...
+		ARG 2 nextScreen
+	METHOD <init> (Lnet/minecraft/network/chat/Component;Lnet/minecraft/client/gui/screens/Screen;)V
+		ARG 1 error
+		ARG 2 nextScreen
+	METHOD <init> (Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/client/gui/screens/Screen;)V
+		ARG 1 line1
+		ARG 2 line2
+		ARG 3 nextScreen
+	METHOD errorMessage (Lcom/mojang/realmsclient/exception/RealmsServiceException;)V
+		ARG 1 exception
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen.mapping]: `errorMessage` method param should be consistent with constructor param: I would suggest calling them both `exception` or `serviceException`.

File: data/com/mojang/realmsclient/gui/screens/RealmsPendingInvitesScreen.mapping [hunk 17]
```diff
@@ -8,6 +10,11 @@ CLASS com/mojang/realmsclient/gui/screens/RealmsPendingInvitesScreen
 		ARG 2 mouseX
 		ARG 3 mouseY
 		ARG 4 partialTicks
+	METHOD renderMousehoverTooltip (Lcom/mojang/blaze3d/vertex/PoseStack;Lnet/minecraft/network/chat/Component;II)V
+		ARG 1 poseStack
+		ARG 2 tooltip
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/gui/screens/RealmsPendingInvitesScreen.mapping]: Passed field name is `toolTip`. 

```suggestion
		ARG 2 toolTip
```

File: data/com/mojang/realmsclient/gui/screens/RealmsSettingsScreen.mapping [hunk 18]
```diff
@@ -1,4 +1,7 @@
 CLASS com/mojang/realmsclient/gui/screens/RealmsSettingsScreen
+	METHOD <init> (Lcom/mojang/realmsclient/gui/screens/RealmsConfigureWorldScreen;Lcom/mojang/realmsclient/dto/RealmsServer;)V
+		ARG 1 configureScreen
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/gui/screens/RealmsSettingsScreen.mapping]: Should match field name. 

```suggestion
		ARG 1 configureWorldScreen
```

File: data/com/mojang/realmsclient/gui/screens/RealmsSubscriptionInfoScreen.mapping [hunk 19]
```diff
@@ -1,4 +1,10 @@
 CLASS com/mojang/realmsclient/gui/screens/RealmsSubscriptionInfoScreen
+	METHOD <init> (Lnet/minecraft/client/gui/screens/Screen;Lcom/mojang/realmsclient/dto/RealmsServer;Lnet/minecraft/client/gui/screens/Screen;)V
+		ARG 1 lastScreen
+		ARG 2 serverData
+		ARG 3 mainScreen
+	METHOD getSubscription (J)V
+		ARG 1 worldId
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/gui/screens/RealmsSubscriptionInfoScreen.mapping]:
```suggestion
		ARG 1 serverId
```

File: data/com/mojang/realmsclient/gui/screens/RealmsUploadScreen.mapping [hunk 20]
```diff
@@ -1,4 +1,22 @@
 CLASS com/mojang/realmsclient/gui/screens/RealmsUploadScreen
+	METHOD <init> (JILcom/mojang/realmsclient/gui/screens/RealmsResetWorldScreen;Lnet/minecraft/world/level/storage/LevelSummary;Ljava/lang/Runnable;)V
+		ARG 1 worldId
+		ARG 3 slotId
+		ARG 4 lastScreen
+		ARG 5 selectedLevel
+		ARG 6 callback
+	METHOD addFileToTarGz (Lorg/apache/commons/compress/archivers/tar/TarArchiveOutputStream;Ljava/lang/String;Ljava/lang/String;Z)V
+		ARG 1 tarArchiveOutputStream
+		ARG 2 toGzip
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/gui/screens/RealmsUploadScreen.mapping]: Argument is for a path; `path` or `pathname`.

File: data/com/mojang/realmsclient/gui/screens/RealmsUploadScreen.mapping [hunk 21]
```diff
@@ -8,3 +26,9 @@ CLASS com/mojang/realmsclient/gui/screens/RealmsUploadScreen
 		ARG 2 mouseX
 		ARG 3 mouseY
 		ARG 4 partialTicks
+	METHOD setErrorMessage ([Lnet/minecraft/network/chat/Component;)V
+		ARG 1 errorMessage
+	METHOD tarGzipArchive (Ljava/io/File;)Ljava/io/File;
+		ARG 1 toGzip
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/gui/screens/RealmsUploadScreen.mapping]:
```suggestion
		ARG 1 file
```

File: data/com/mojang/realmsclient/util/RealmsUtil.mapping [hunk 22]
```diff
@@ -0,0 +1,9 @@
+CLASS com/mojang/realmsclient/util/RealmsUtil
+	METHOD convertToAgePresentation (J)Ljava/lang/String;
+		ARG 0 time
+	METHOD convertToAgePresentationFromInstant (Ljava/util/Date;)Ljava/lang/String;
+		ARG 0 date
+	METHOD getTextures (Ljava/lang/String;)Ljava/util/Map;
+		ARG 0 uuid
+	METHOD uuidToName (Ljava/lang/String;)Ljava/lang/String;
+		ARG 0 uuid
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/util/RealmsUtil.mapping]: IMO it's a little unclear what UUID these methods take without looking at the source. I would suggest `profileUuid` or `playerUuid`.

File: data/com/mojang/realmsclient/util/task/ResettingGeneratedWorldTask.mapping [hunk 23]
```diff
@@ -0,0 +1,6 @@
+CLASS com/mojang/realmsclient/util/task/ResettingGeneratedWorldTask
+	METHOD <init> (Lcom/mojang/realmsclient/util/WorldGenerationInfo;JLnet/minecraft/network/chat/Component;Ljava/lang/Runnable;)V
+		ARG 1 generationInfo
+		ARG 2 worldId
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/util/task/ResettingGeneratedWorldTask.mapping]:
```suggestion
		ARG 2 serverId
```

File: data/com/mojang/realmsclient/util/task/ResettingWorldTask.mapping [hunk 24]
```diff
@@ -0,0 +1,8 @@
+CLASS com/mojang/realmsclient/util/task/ResettingWorldTask
+	METHOD <init> (JLnet/minecraft/network/chat/Component;Ljava/lang/Runnable;)V
+		ARG 1 serverId
+		ARG 3 title
+		ARG 4 callback
+	METHOD sendResetRequest (Lcom/mojang/realmsclient/client/RealmsClient;J)V
+		ARG 1 client
+		ARG 2 worldId
```
Comment (Harleyoc1) [inline data/com/mojang/realmsclient/util/task/ResettingWorldTask.mapping]:
```suggestion
		ARG 2 serverId
```

**Review by Harleyoc1 — CHANGES_REQUESTED** [review]

File: data/com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen.mapping (hunk 16 above)
Comment (FlowArg) [reply data/com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen.mapping]: IMO `serviceException` is more precise so it's better than just `exception`.

File: data/com/mojang/realmsclient/util/RealmsUtil.mapping (hunk 22 above)
Comment (FlowArg) [reply data/com/mojang/realmsclient/util/RealmsUtil.mapping]: Good idea, I think `profileUuid` is better.

**Review by Harleyoc1 — APPROVED** [review]

**Review by sciwhiz12 — APPROVED** [review]

**Pull request merged**